src/Middleware/Session/src/DistributedSession.cs

```diff
@@ -262,12 +262,6 @@ public async Task LoadAsync(CancellationToken cancellationToken = default)
     /// <inheritdoc />
     public async Task CommitAsync(CancellationToken cancellationToken = default)
     {
-        if (!IsAvailable)
```

Comment:
```
// use field to avoid loading the session via IsAvailable property
if (!_isAvailable)
```
?

Reply:
Not sure we want that; if `_isAvailable` wasn't set, it could be that we never attempted to load it. The current code attempts to load in that case.

Reply:
Why would we remove this check in `CommitAsync` but not in `Set`?

Reply:
`Set` always depends on `IsAvailable`, but `CommitAsync` might not need to load the session if `_isModified` isn't true. The `IsAvailable` check was added in `CommitAsync` during the nullability work but I don't see why it's necessary now.

Reply:
I see. And it could force sync-over-async as well. Yuck!